source/header_cell.ts

```diff
@@ -0,0 +1,105 @@
+import {Sorting} from './sorting';
+import {Filter} from './filter';
+
+/** Header of a column of a table. */
+export class HeaderCell {
+
+  /** Constructs a new HeaderCell.
+    * @param name - The name to display if there is enough available space.
+    * @param shortName - The name to display if there is not enough space.
+    * @param isSortable - Indicates if the column is sortable.
+    * @param sorting - The current sorting of the column.
+    */
+  constructor(name: string, shortName: string, isSortable: boolean = true,
+      sorting: Sorting = Sorting.NONE) {
+    this.#name = name;
+    this.#shortName = shortName;
+    if(isSortable) {
+      this.#sorting = new HeaderCell.Sortable(sorting);
+    } else {
+      this.#sorting = new HeaderCell.Unsortable();
+    }
+    this.#filterable = new HeaderCell.Unfilterable();
+  }
+
+  /** Get the full name of the header. */
+  get name(): string {
+    return this.#name;
+  }
+
+  /** Get the condensed name of the header. */
+  get shortName(): string {
+    return this.#shortName;
+  }
+
+  /** Get sortable object from the header. */
+  get sortOrder(): Sorting {
+    return this.#sorting.sortOrder;
+  }
+
+  /** Get sortable object from the header. */
+  get sortable(): HeaderCell.Sortable | HeaderCell.Unsortable {
+    return this.#sorting;
+  }
+
+  /** Get sortable object from the header. */
+  get filter(): Filter {
+    return this.#filterable.filter;
+  }
+
+  /** Get filterable object from the header. */
+  get filterable(): HeaderCell.Unfilterable {
+    return this.#filterable;
+  }
+
+  #name: string;
+  #shortName: string;
+  #sorting: HeaderCell.Sortable | HeaderCell.Unsortable;
+  #filterable: HeaderCell.Unfilterable;
+}
+
+export namespace HeaderCell {
+
+  /** Sortable header. */
+  export class Sortable {
```

Comment:
Use nouns for type names.

Reply:
My intention was for Sortable to be a noun like Iterable. I'll rename it to something clearer.

Reply:
Yeah, looking it over it doesn't look like this class does anything. You may want to consider removing it altogether. It has nothing but a getter and a setter, which makes it look like a really expensive way of wrapping a variable. What was the use case for it?